functions/Find-DbaDisabledIndex.ps1

```diff
@@ -0,0 +1,180 @@
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+        .LINK
+            https://dbatools.io/Find-DbadisabledIndex
+
+        .EXAMPLE
+            Find-DbadisabledIndex -SqlInstance sql2005
+
+            Generates the SQL statements to drop the selected disabled indexes on server "sql2005".
+
+        .EXAMPLE
+            Find-DbadisabledIndex -SqlInstance sqlserver2016 -SqlCredential $cred
+
+            Generates the SQL statements to drop the selected disabled indexes on server "sqlserver2016", using SQL Authentication to connect to the database.
+
+        .EXAMPLE
+            Find-DbadisabledIndex -SqlInstance sqlserver2016 -Database db1, db2
+
+            Generates the SQL Statement to to drop selected indexes in databases db1 & db2 on server "sqlserver2016".
+
+        .EXAMPLE
+            Find-DbadisabledIndex -SqlInstance sqlserver2016
+
+            Generates the SQL statements to drop selected indexes on all user databases.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    Param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]
+        $SqlCredential,
+        [Alias("Databases")]
+        [object[]]$Database,
+        [object[]]$ExcludeDatabase,
+        [switch]$NoClobber,
+        [switch]$Append,
+        [switch][Alias('Silent')]$EnableException
+    )
+
+    begin {
+        # 
+        $disabledQuery = "
+        SELECT DB_NAME() AS 'DatabaseName'
+        ,s.name AS 'SchemaName'
+        ,t.name AS 'TableName'
+        ,i.object_id AS ObjectId
+        ,i.name AS 'IndexName'
+        ,i.index_id as 'IndexId'
+        ,i.type_desc as 'TypeDesc'
+        FROM sys.tables t
+        JOIN sys.schemas s
+            ON t.schema_id = s.schema_id
+        JOIN sys.indexes i
+            ON i.object_id = t.object_id
+        WHERE i.is_disabled = 1"
+
+        Write-Message -Level Output -Message "Attempting to connect to Sql Server."
+        $server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
+    }
+    process {
+        if (Test-FunctionInterrupt) { return }
+
+        if ($server.VersionMajor -lt 9) {
+            Stop-Function -Message "This function does not support versions lower than SQL Server 2005 (v9)."
+            return
+        }
+
+        $lastRestart = $server.Databases['tempdb'].CreateDate
+        $endDate = Get-Date -Date $lastRestart
+        $diffDays = (New-TimeSpan -Start $endDate -End (Get-Date)).Days
+
+        <#
+            Validate if server version is:
+                - sql 2012 and if have SP3 CU3 (Build 6537) or higher
+                - sql 2014 and if have SP2 (Build 5000) or higher
+            If the major version is the same but the build is lower, throws the message
+        #>
+        if ($pipedatabase.Length -gt 0) {
+            $database = $pipedatabase.name
```

Comment:
This variable would be best called `$databases` to differentiate it from the parameter on the command.

Reply:
Don't know that we have in any particular guide but I try to not overwrite the parameter for the command, so in this case `$Database` should only ever exist as what is passed in.